### PR TITLE
Bump google-api-python-client from ==1.5.1 to >=1.6.2

### PR DESCRIPTION
**Pull request by jpvelez**
Hello! 

We're trying to install datalab internally at Spotify and running into dependency collisions.

By pinning google-api-python-client to 1.5.1, `datalab` package ends up install `uritemplate=0.6`:

```
Collecting uritemplate<1,>=0.6 (from google-api-python-client==1.5.1->datalab)
```

uritemplate is up to 3.0.x now. 0.6 is an old version of the package which is incompatible with dozens of packages in anaconda distro.

Could we bump the `google-api-python-client` dependency to `>=1.6.2`? Those folks upgrade `uritemplate` to 3.0.0 in version `1.6.2`: https://github.com/google/google-api-python-client/blob/master/setup.py#L69

### CODE REVIEW

**Comment by jpvelez** [comment]
To be extra explicit, here's what happening.

Pip tries to install `datalab`. It finds the existing install of uritemplate that ships with the `conda` command line tool (not even the full blown anaconda installation:

```
Found existing installation: uritemplate 3.0.0
    Uninstalling uritemplate-3.0.0:
      Successfully uninstalled uritemplate-3.0.0
```

It removes 3.0.0 and successfully installs 0.6.0:

```
Successfully installed cycler-0.10.0 datalab-1.0.1 funcsigs-1.0.2 future-0.16.0 futures-3.0.5 gapic-google-logging-v2-0.9.3 gapic-google-pubsub-v1-0.9.3 google-api-python-client-1.5.1 google-cloud-0.19.0 google-gax-0.13.0 googleapis-common-protos-1.5.2 grpc-google-logging-v2-0.9.3 grpc-google-pubsub-v1-0.9.3 grpcio-1.4.0 httplib2-0.9.2 ipykernel-4.5.2 matplotlib-2.0.2 mock-2.0.0 numpy-1.13.1 oauth2client-2.2.0 pandas-0.20.3 pandas-profiling-1.4.0 pbr-3.1.1 plotly-1.12.5 ply-3.8 protobuf-3.3.0 psutil-4.3.0 pyasn1-modules-0.0.11 pyparsing-2.2.0 python-dateutil-2.5.0 pytz-2017.2 pyyaml-3.11 requests-2.9.1 rsa-3.4.2 scikit-learn-0.17.1 scipy-0.19.1 seaborn-0.7.0 subprocess32-3.2.7 uritemplate-0.6
```

But it appears that uritemplate=3.0.0 is a dependency of pip itself, conda's version anyway, because immediately after this "Successful install" message this happens:


```
Traceback (most recent call last):
  File "/opt/conda/bin/pip", line 6, in <module>
    sys.exit(pip.main())
  File "/opt/conda/lib/python2.7/site-packages/pip/__init__.py", line 249, in main
    return command.main(cmd_args)
  File "/opt/conda/lib/python2.7/site-packages/pip/basecommand.py", line 252, in main
    pip_version_check(session)
  File "/opt/conda/lib/python2.7/site-packages/pip/utils/outdated.py", line 102, in pip_version_check
    installed_version = get_installed_version("pip")
  File "/opt/conda/lib/python2.7/site-packages/pip/utils/__init__.py", line 838, in get_installed_version
    working_set = pkg_resources.WorkingSet()
  File "/opt/conda/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 644, in __init__
    self.add_entry(entry)
  File "/opt/conda/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 700, in add_entry
    for dist in find_distributions(entry, True):
  File "/opt/conda/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 1949, in find_eggs_in_zip
    if metadata.has_metadata('PKG-INFO'):
  File "/opt/conda/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 1463, in has_metadata
    return self.egg_info and self._has(self._fn(self.egg_info, name))
  File "/opt/conda/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 1823, in _has
    return zip_path in self.zipinfo or zip_path in self._index()
  File "/opt/conda/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 1703, in zipinfo
    return self._zip_manifests.load(self.loader.archive)
  File "/opt/conda/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 1643, in load
    mtime = os.stat(path).st_mtime
OSError: [Errno 2] No such file or directory: '/opt/conda/lib/python2.7/site-packages/uritemplate-3.0.0-py2.7.egg'
```

As far as I can tell, as pip moves to wrap up the installation of `datalab`, it tries to use `uritemplate-3.0.0-py2.7`, and can't.

In other words, this may be an incompatibility between `conda` and `datalab`, which is =-(

**Comment by yebrahim** [comment]
Thanks for the change @jpvelez. Why not pin the library to the new version `1.6.2` though?

**Comment by jpvelez** [comment]
@yebrahim so we don't run into this problem again in the future, I suppose? Happy to pin it - I'm a bit new to python packaging, actually, and would like to understand better your strategy of pinning all dependencies to a certain version vs. always pulling the latest?

**Comment by yebrahim** [comment]
@jpvelez we do this so that future releases of our dependencies don't break us. We upgrade our dependencies manually after testing that they work well.

Thanks for the change.

**Comment by jpvelez** [comment]
@yebrahim makes sense. Any chance you could do a release on PyPi so we can install this latest version? If not, how do you recommend we install - directly from Github via pip?

**Comment by yebrahim** [comment]
@jpvelez yes we should be making a PyPI release very soon. If you need it before then you can just clone and install the github repo, this is actually what datalab does at the moment.

**Comment by jpvelez** [comment]
@yebrahim excellent, thank you!
